src/prpy/planning/vectorfield.py

```diff
@@ -526,11 +528,10 @@ def fn_callback(t, q):
                 if path.GetNumWaypoints() == 1:
                     checks = [(t, q)]
                 else:
-                    # TODO: This should start at t_check. Unfortunately, a bug
-                    # in GetCollisionCheckPts causes this to enter an infinite
-                    # loop.
-                    checks = GetCollisionCheckPts(robot, path,
-                                                  include_start=False)
+                    # TODO: This will recheck the entire trajectory
+                    #  Ideally should just check the new portion of the trajectory
+                    checks = GetLinearCollisionCheckPts(robot, path,
+                            norm_order=norm_order, sampling_func=sampling_func)
```

Comment:
Unfortunately, this isn't enough. `GetLinearCollisionCheckPts` retimes the trajectory internally and returns timestamps on the retimed trajectory. This means that the value of `t_check` below does not correspond to the original trajectory.

This change causes `VectorFieldPlanner` to fail a unit test about constraint deviation because the trajectory is being sampled at arbitrary (incorrect) times.